### DISCUSSION TITLE
Problem with projection fields with parameter on 8.0.0

### DISCUSSION THREAD

**Discussion by dirkvanroy80** [Netflix/dgs-framework#1690] [Q&A]
I updated to the latest versions of the dgs codegen project. I changed all my ProjectionRoot() to ProjectionRoot<>() to get it working, but there is one projection where one of the fields requires a parameter. When I use this, the parent type seems to get lost, so if I use parent(), I get an Object type as return value and cannot select other fields any more.
example:

new TestProjectionRoot<>()
  .subObject()
  .canFollow(parameter)
  .parent() -> returns Object instead of a TestProjectionRoot.
  .nextSubObject() -> doesn't work

Is this a bug? Or am I doing something wrong?

**Comment by srinivasankavitha** [comment]
Not sure, depends on your schema. You could try navigating up the parent one more level or go to the root and navigate down depending on where the field is located.
